src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -197,10 +196,10 @@ public void run() {
                                         } finally {
                                             if (!expriment_exists) {
                                                 LOGGER.info("Processing experiment {}", jobData.getProcessed_experiments());
-                                                jobData.setProcessed_experiments(jobData.getProcessed_experiments() + 1);
+                                                jobData.incrementProcessed_experiments();
                                             }
                                             synchronized (new Object()) {
```

Comment:
The sychronized object has to be the same for every thread ...  Suggest you use `jobData` or an explicit lock variable inside jobData

Reply:
> The sychronized object has to be the same for every thread ... Suggest you use `jobData` or an explicit lock variable inside jobData

The [processed_experiments](https://github.com/kruize/autotune/blob/5b16e039e21048a702869f6e7c32cb30eccea94e/src/main/java/com/autotune/analyzer/serviceObjects/BulkJobStatus.java#L45) variable is shared among threads and is already an AtomicInteger, which ensures thread-safe increment operations without requiring explicit synchronization.

If you think synchronization is still necessary, could you please elaborate on the specific scenario where thread safety might fail with the current implementation? Alternatively, I can explore synchronizing on a lock object within jobData as suggested. 

Let me know your thoughts.

Reply:
I am talking about the usage of `synchornized` keyword in lines 201 and 231 with (new Object()) as the variable. Do we need these still? If so we need to use `jobData` or an explicit lock as the parameter

Reply:
Thank you for pointing this out. Using new Object() as the synchronized monitor indeed creates a new instance each time, which defeats the purpose of synchronization. I agree that it would be more appropriate to synchronize on a shared, consistent object.  Can I consider this is different PR as we may move into DB store this code may nt be valid there ?

Reply:
Since that code is not working, lets fix it once and for all.

Reply:
Changes incorporated


Seems working fine with 800 experiments. Logs attached


```
#######################################
# Kruize Demo Setup on openshift 
#######################################

#######################################
1. Cloning autotune git repos
done
#######################################


#######################################
6. Installing Kruize
route/kruize exposed
route/kruize-ui-nginx-service exposed
route.route.openshift.io/kruize annotated

######################################################
#     Install default metric profile
######################################################

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 22663    0   192  100 22471    554  64856 --:--:-- --:--:-- --:--:-- 65500

Running bulk_demo.py...
bulk_demo.py -c openshift
IP =  kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com

KRUIZE URL =  http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com

#######################################
Invoking bulk service
#######################################

URL =  http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com/bulk
Response -  {'job_id': '5dfcf789-a9d3-4eca-b6fc-3114c474428b'}

#######################################
Querying job status in a loop
#######################################

Experiments: processed / Total -  26 / 851
Experiments: processed / Total -  84 / 851
Experiments: processed / Total -  131 / 851
Experiments: processed / Total -  198 / 851
Experiments: processed / Total -  251 / 851
Experiments: processed / Total -  290 / 851
Experiments: processed / Total -  339 / 851
Experiments: processed / Total -  412 / 851
Experiments: processed / Total -  468 / 851
Experiments: processed / Total -  544 / 851
Experiments: processed / Total -  601 / 851
Experiments: processed / Total -  680 / 851
Experiments: processed / Total -  727 / 851
Experiments: processed / Total -  780 / 851
Experiments: processed / Total -  841 / 851
Experiments: processed / Total -  851 / 851

#######################################
Bulk Job Completed! Fetching the processed experiments and listing recommendations
#######################################

Experiment names written to experiment_list.txt
Recommendations for 1 container is available in recommendations_data.json
List of all experiments are available in experiment_list.txt

######################################################


#######################################
#             Access Kruize           #
#######################################
ℹ️  Access kruize UI at http://kruize-ui-nginx-service-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com
🔖 To explore further, access kruize UI to list and create experiments, and to view or generate recommendations!
ℹ️  For kruize CLI commands, refer to the end of ./kruize-bulk-demo.log

-------------------------------------------
          Access Kruize Interface          
-------------------------------------------
Kruize UI:
   Open the URL in your browser: http://kruize-ui-nginx-service-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com

-------------------------------------------
          CLI Commands for Kruize          
-------------------------------------------
1. Create Experiment:
   curl -X POST http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com/createExperiment -d @./experiments/{experiment}.json

2. Generate Recommendations for an Experiment:
   curl -X POST http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com/generateRecommendations?experiment_name={experiment_name}

3. List Recommendations for an Experiment:
   curl -X POST http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com/listRecommendations?experiment_name={experiment_name}

4. List all Experiments:
   curl http://kruize-openshift-tuning.apps.kruize2.lab.psi.pnq2.redhat.com/listExperiments
-------------------------------------------

For kruize local documentation, refer https://github.com/kruize/autotune/blob/master/design/KruizeLocalAPI.md
For bulk documentation, refer https://github.com/kruize/autotune/blob/master/design/BulkAPI.md
```